src/ruby/ext/symengine/ruby_basic.c

```diff
@@ -0,0 +1,109 @@
+#include "ruby_basic.h"
+
+static void cbasic_free(void *ptr){
+    basic_struct *basic_ptr = ptr;
+    basic_free(basic_ptr);
+}
+
+static VALUE cbasic_alloc(VALUE klass){
```

Comment:
This line generates the following error:
```
ruby_basic.c:8:14: error: ‘cbasic_alloc’ defined but not used [-Werror=unused-function]
 static VALUE cbasic_alloc(VALUE klass){
              ^
```
I think that's because the function is declared `static`, so it is local to this file. I don't think you should declare it as static.